library/core/src/main/java/com/google/android/exoplayer2/extractor/mkv/MatroskaExtractor.java

```diff
@@ -1931,4 +1947,68 @@ private static boolean parseMsAcmCodecPrivate(ParsableByteArray buffer) throws P
 
   }
 
+  private class TrueHDSampleHolder {
+
+    private int counter;
+    private int samplesSize;
+    private long timeUs;
+    private @C.BufferFlags int blockFlags;
+    private boolean gotSyncFrame;
+    private boolean letFirstIn;
+    private Track track;
+
+    TrueHDSampleHolder() {
+      reset();
+    }
+
+    private boolean isFirstSampleCheck() {
+      return counter == 1;
+    }
+
+    private void resetCounters() {
+      track = null;
+      counter = 0;
+      samplesSize = 0;
+      timeUs = 0;
+      blockFlags = 0;
+    }
+
+    void reset() {
+      gotSyncFrame = false;
+      letFirstIn = true;
+      resetCounters();
+    }
+
+    void testSample(ExtractorInput input, int size, @C.BufferFlags int blockFlags, Track track)
+            throws IOException, InterruptedException {
+      samplesSize += size;
+      ++counter;
+      if (isFirstSampleCheck())
+        this.blockFlags = blockFlags;
+      if (!gotSyncFrame) {
+        byte[] bytes = new byte[12];
+        input.peekFully(bytes, 0, 12);
+        input.resetPeekPosition();
+        if (0 < Ac3Util.parseTrueHDSyncframeAudioSampleCount(ByteBuffer.wrap(bytes).order(ByteOrder.LITTLE_ENDIAN))) {
```

Comment:
Could parseTrueHDSyncframeAudioSampleCount take a byte[] to avoid wrapping every time?

Reply:
Yep